Makefile

```diff
@@ -5,40 +5,33 @@ SELF_MAKE := $(lastword $(MAKEFILE_LIST))
 
 PKG_REPO = testpypi
 PKG_INCREMENT := patch
-PKG_SET := tools/c7n_gcp tools/c7n_kube tools/c7n_openstack tools/c7n_mailer tools/c7n_logexporter tools/c7n_policystream tools/c7n_trailcreator tools/c7n_org tools/c7n_sphinxext tools/c7n_awscc tools/c7n_tencentcloud tools/c7n_azure tools/c7n_oci tools/c7n_terraform
+PKG_SET := tools/c7n_gcp tools/c7n_kube tools/c7n_openstack tools/c7n_mailer tools/c7n_logexporter tools/c7n_policystream tools/c7n_trailcreator tools/c7n_org tools/c7n_sphinxext tools/c7n_awscc tools/c7n_tencentcloud tools/c7n_azure tools/c7n_oci tools/c7n_terraform tools/c7n_left
 
 FMT_SET := tools/c7n_left tools/c7n_mailer tools/c7n_oci tools/c7n_kube tools/c7n_awscc
 
-PLATFORM_ARCH := $(shell python3 -c "import platform; print(platform.machine())")
-PLATFORM_OS := $(shell python3 -c "import platform; print(platform.system())")
-PY_VERSION := $(shell python3 -c "import sys; print('%s.%s' % (sys.version_info.major, sys.version_info.minor))")
-
 COVERAGE_TYPE := html
 ARGS :=
 IMAGE := c7n
 IMAGE_TAG := latest
 
-# we distribute tfparse binary wheels for 3.10+
-ifneq "$(findstring 3.1, $(PY_VERSION))" ""
-    PKG_SET := tools/c7n_left $(PKG_SET)
-endif
-
-
 ###
 # Common developer targets
 
 install:
-	@if [[ -z "$(VIRTUAL_ENV)" ]]; then echo "Create and Activate VirtualEnv First, ie. python3 -m venv .venv && source .venv/bin/activate"; exit 1; fi
-	poetry install --with addons
-	for pkg in $(PKG_SET); do echo "Install $$pkg" && cd $$pkg && poetry install --all-extras && cd ../..; done
+# extras are for c7n_mailer, separate lint from dev for ci
+	uv sync --all-packages --locked \
+	    --group dev \
+	    --group addons \
+	    --group lint \
+            --extra gcp --extra azure
```

Comment:
FWIW I get something like this trying to run `make install` with an existing venv active:

> warning: `VIRTUAL_ENV=/home/aj/.venvs/c7n-dev` does not match the project environment path `.venv` and will be ignored; use `--active` to target the active environment instead

That seems like a pretty clear message with a pretty clear resolution (deactivate the current venv), but noting in case others have their venv autopilot on and that's a case we want to explicitly address down the road.